spec/compiler/interpreter/lib_spec.cr

```diff
@@ -0,0 +1,57 @@
+{% skip_file if flag?(:without_interpreter) %}
+require "./spec_helper"
+require "../loader/spec_helper"
+
+describe Crystal::Repl::Interpreter do
+  context "variadic calls" do
+    before_all do
+      FileUtils.mkdir_p(SPEC_CRYSTAL_LOADER_LIB_PATH)
+      build_c_dynlib(compiler_datapath("interpreter", "sum.c"))
+    end
+
+    it "promotes float" do
+      interpret(<<-CR).should eq 3.5
+        @[Link(ldflags: "-L#{SPEC_CRYSTAL_LOADER_LIB_PATH} -lsum")]
+        lib LibSum
+          fun sum_float(count : Int32, ...) : Float32
+        end
+
+        LibSum.sum_float(2, 1.2_f32, 2.3_f32)
+        CR
+    end
+
+    it "promotes int" do
+      interpret(<<-CR).should eq 5
+        @[Link("#{SPEC_CRYSTAL_LOADER_LIB_PATH}/libsum.so")]
```

Comment:
Can we use `Crystal::Loader::SHARED_LIBRARY_EXTENSION` here? This fails to run on my mac.

Reply:
I just messed this up. The best solution is to use `ldflags: "-L#{SPEC_CRYSTAL_LOADER_LIB_PATH} -lsum"`. I already did that in the first example, but not the other two. Fixed.

Reply:
It worked, thanks! Which means I'll probably be able to reproduce the global var proc thing locally, and maybe fix it.